Automattic/jetpack
# API: 3rd-party: Add tools for getting module overrides; Add to API

As part of my work on #8800, I need to know which modules are being forced on and which are being forced off. With that information, it should be relatively easy to modify the UI to let a user know that an option can not be modified.

Updating the UI to reflect whether an option can be updated is especially important in the context of a hosting partner forcing modules on/off. 

This PR also adds the `override` key to each module result returned in the API. This will allow us to support module overrides on both WPCOM and in the Jetpack UI.

To test:

- Review code and ensure testing logic is sound
- `phpunit --filter=Jetpack_Modules_Overrides`
- Optionally functionally test in `wp shell`
- Checkout D10224-code on WPCOM, sandbox API and query the following:
    - `GET /sites/$site/jetpack/modules`
    - `GET /sites/$site/jetpack/modules/$module_slug`
- Ensure that both include the `override` key for modules
- Add the following to an integration plugin:
    ```
    add_filter( 'option_jetpack_active_modules', 'jetpack_test_module_override' );
    function jetpack_test_module_override( $modules ) {
    	$found = array_search( 'sitemaps', $modules );
    	if ( $found ) {
    		unset( $modules[ $found ] );
    	}
    
    	return array_values( array_merge( $modules, array( 'photon', 'lazy-images' ) ) );
    }
    ```
- Run `GET /sites/$site/jetpack/modules` and ensure that the override for `photon` and `lazy-images` is `active`, the override for `sitemaps` is `inactive`, and all others have an empty value.

cc @oskosk since you mentioned you looked at this before in #8800.
